pkg/platform/runtime/buildscript/file.go

```diff
@@ -9,37 +9,57 @@ import (
 	"github.com/ActiveState/cli/internal/errs"
 	"github.com/ActiveState/cli/internal/fileutils"
 	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/pkg/localcommit"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/model"
 	"github.com/ActiveState/cli/pkg/platform/runtime/buildexpression"
+	"github.com/ActiveState/cli/pkg/projectfile"
 )
 
-type DoesNotExistError struct{ error }
-
-func IsDoesNotExistError(err error) bool {
-	return errs.Matches(err, &DoesNotExistError{})
+func NewScriptFromProjectDir(dir string, auth *authentication.Auth) (*Script, error) {
+	return newScriptFromFile(filepath.Join(dir, constants.BuildScriptFileName), auth)
 }
 
-func NewScriptFromProjectDir(dir string) (*Script, error) {
-	return newScriptFromFile(filepath.Join(dir, constants.BuildScriptFileName))
-}
+func newScriptFromFile(path string, auth *authentication.Auth) (*Script, error) {
+	if data, err := fileutils.ReadFile(path); err == nil {
+		return NewScript(data)
+	} else if !errors.Is(err, os.ErrNotExist) {
+		return nil, errs.Wrap(err, "Could not read build script")
+	}
 
-func newScriptFromFile(path string) (*Script, error) {
-	data, err := fileutils.ReadFile(path)
+	logging.Debug("Build script does not exist. Creating one.")
+	dir := filepath.Dir(path)
+	pjf, err := projectfile.FromPath(dir)
```

Comment:
In every scenario this is called you are passing the dir from the project package. Then you are constructing the projectfile based on this dir. Seems more productive if we can just pass the actual project instance.